### PR TITLE
lnwallet/test: reproduce test vectors in commit and htlc tx test 

### PR DESCRIPTION
**Pull request by joostjager**
This PR changes the verification of our code against the spec test vectors to use a more black box approach. It exercises the channel state machine via its external interface as much as possible, making this test more robust. A consequence is that the test now runs from the 'root' data from which the test vectors are also derived, meaning that more code is covered too.

Running from the root data is also a preparation for _producing_ test vectors for the new anchor commitment format. This will be a matter of changing the channel type and recording the produced commitment and htlc txes.

Previously the success transaction was skipped during verification. With this commit, the proper preimage insertion is carried out, allowing the success tx to be checked too.

Finally two test cases are removed that are invalid in `lnd` and probably in the protocol in general.

### CODE REVIEW

File: lnwallet/transactions_test.go [hunk 1]
```diff
@@ -649,218 +446,189 @@ var testCases = []testCase{
...
+		witnessScript := successTx.TxIn[0].Witness[4]
+		var hash160 [20]byte
+		copy(hash160[:], witnessScript[69:69+20])
+		preimage := hash160map[hash160]
+		successTx.TxIn[0].Witness[3] = preimage[:]
+		secondLevelTxes[r.HtlcPoint().Index] = successTx
+	}
+	for _, r := range forceCloseSum.HtlcResolutions.OutgoingHTLCs {
+		secondLevelTxes[r.HtlcPoint().Index] = r.SignedTimeoutTx
+	}
```
Comment (cfromknecht) [inline lnwallet/transactions_test.go:581]: add an assertion on the size of `secondLevelTxes` to ensure nothing was overwritten?

File: lnwallet/transactions_test.go [hunk 2]
```diff
@@ -649,218 +446,189 @@ var testCases = []testCase{
...
-			if err != nil {
-				t.Fatalf("Failed to parse serialized tx: %v", err)
-			}
+	// Assert that the commitment transaction itself is as expected.
+	var txBytes bytes.Buffer
+	require.NoError(t, forceCloseSum.CloseTx.Serialize(&txBytes))
 
-			htlcResolution := htlcResolutions.OutgoingHTLCs[resolutionIdx]
-			resolutionIdx++
+	fmt.Printf("commit tx: %x\n", txBytes.Bytes())
```
Comment (cfromknecht) [inline lnwallet/transactions_test.go:561]: stray `Printf`

File: lnwallet/transactions_test.go [hunk 3]
```diff
@@ -649,218 +446,189 @@ var testCases = []testCase{
...
+		var b bytes.Buffer
+		err := tx.Serialize(&b)
+		require.NoError(t, err)
+
+		require.Equal(
+			t,
+			test.htlcDescs[i].resolutionTxHex,
+			hex.EncodeToString(b.Bytes()),
+		)
+		fmt.Printf("second-level tx %v: %x\n", idx, b.Bytes())
```
Comment (cfromknecht) [inline lnwallet/transactions_test.go:605]: same here

File: lnwallet/transactions_test.go [hunk 4]
```diff
@@ -649,218 +446,189 @@ var testCases = []testCase{
...
-
-		// This is the remote party's signature over the commitment
-		// transaction which is included in the commitment tx's witness
-		// data.
-		channelState.LocalCommitment.CommitSig, err = hex.DecodeString(test.remoteSigHex)
-		if err != nil {
-			t.Fatalf("Case %d: Failed to parse serialized signature: %v",
-				i, err)
-		}
+	fmt.Printf("Remote sig: %x\n", remoteSig.ToSignatureBytes())
```
Comment (cfromknecht) [inline lnwallet/transactions_test.go:537]: here too

File: lnwallet/transactions_test.go [hunk 5]
```diff
@@ -649,218 +446,189 @@ var testCases = []testCase{
 func TestCommitmentAndHTLCTransactions(t *testing.T) {
 	t.Parallel()
 
-	tc, err := newTestContext()
-	if err != nil {
-		t.Fatal(err)
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			testVectors(t, test) // nolint:scopelint
```
Comment (cfromknecht) [inline lnwallet/transactions_test.go:425]: bind `test := test` to avoid nolint?

Comment (joostjager) [reply lnwallet/transactions_test.go:425]: Fixed. But I was wondering why it complains? `test` isn't a reference

File: lnwallet/transactions_test.go (hunk 1 above)
Comment (joostjager) [reply lnwallet/transactions_test.go:581]: Added `storeTx` closure with assertion.

File: lnwallet/transactions_test.go (hunk 2 above)
Comment (joostjager) [reply lnwallet/transactions_test.go]: removed

File: lnwallet/transactions_test.go (hunk 5 above)
Comment (cfromknecht) [reply lnwallet/transactions_test.go]: indeed, i don't think there is actually a problem. i would attribute it to either the linter being overly cautious (not distinguishing between ptr and value) or just a bug

**Pull request merged**